sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java

```diff
@@ -389,23 +414,50 @@ public HttpClient build() {
             = (proxyOptions == null) ? ProxyOptions.fromConfiguration(buildConfiguration, true) : proxyOptions;
 
         if (buildProxyOptions != null) {
-            httpClientBuilder
-                = httpClientBuilder.proxySelector(new OkHttpProxySelector(buildProxyOptions.getType().toProxyType(),
-                    buildProxyOptions::getAddress, buildProxyOptions.getNonProxyHosts()));
+            httpClientBuilder.proxySelector(new OkHttpProxySelector(buildProxyOptions.getType().toProxyType(),
+                buildProxyOptions::getAddress, buildProxyOptions.getNonProxyHosts()));
 
             if (buildProxyOptions.getUsername() != null) {
                 ProxyAuthenticator proxyAuthenticator
                     = new ProxyAuthenticator(buildProxyOptions.getUsername(), buildProxyOptions.getPassword());
 
-                httpClientBuilder = httpClientBuilder.proxyAuthenticator(proxyAuthenticator)
+                httpClientBuilder.proxyAuthenticator(proxyAuthenticator)
                     .addInterceptor(proxyAuthenticator.getProxyAuthenticationInfoInterceptor());
             }
         }
 
         // Set the followRedirects property.
         httpClientBuilder.followRedirects(this.followRedirects);
 
+        // Set the protocol versions.
+        if (this.protocolVersions != null) {
+            httpClientBuilder.protocols(toOkHttpProtocol(this.protocolVersions));
+        } else {
+            httpClientBuilder.protocols(Collections.singletonList(Protocol.HTTP_1_1));
+        }
+
         return new OkHttpAsyncHttpClient(httpClientBuilder.build(),
             getTimeout(responseTimeout, getDefaultResponseTimeout()));
     }
+
+    private static List<Protocol> toOkHttpProtocol(EnumSet<HttpProtocolVersion> protocolVersions) {
+        List<Protocol> protocols = new ArrayList<>(protocolVersions.size());
+        for (HttpProtocolVersion protocolVersion : protocolVersions) {
+            switch (protocolVersion) {
+                case HTTP_1_1:
+                    protocols.add(Protocol.HTTP_1_1);
+                    break;
+
+                case HTTP_2:
+                    protocols.add(Protocol.HTTP_2);
+                    break;
+
+                default:
+                    throw LOGGER.logExceptionAsError(
+                        new IllegalArgumentException("'protocolVersions' must contain HTTP_1_1 (HTTP/1.1)."));
```

Comment:
The error message should be something along the lines of "Unsupported HTTP version. This client supports HTTP 1.1 or HTTP2".